modules/python/test/test_houghcircles.py

```diff
@@ -80,5 +80,54 @@ def test_houghcircles(self):
         self.assertLess(float(len(circles) - matches_counter) / len(circles), .75)
 
 
+    def test_houghcircles_alt(self):
+
+        fn = "samples/data/board.jpg"
+
+        src = self.get_sample(fn, 1)
+        img = cv.cvtColor(src, cv.COLOR_BGR2GRAY)
+        img = cv.medianBlur(img, 5)
+
+        circles = cv.HoughCircles(img, cv.HOUGH_GRADIENT_ALT, 1, 10, np.array([]), 300, 0.9, 1, 30)
+
+        self.assertEqual(circles.shape[0], 1)
+        self.assertEqual(circles.shape[1], 18)
+        self.assertEqual(circles.shape[2], 3)
```

Comment:
What about extra dimension that was there?

Reply:
There was no extra dimension. There was a wrong order of shape.

Wrong shape order (for this sample):
circles.shape[0] = 18
circles.shape[1] = 1
circles.shape[2] = 3

The required shape order:
circles.shape[0] = 1
circles.shape[1] = 18
circles.shape[2] = 3

Reply:
Please use this form:

```
self.assertEqual(circles.shape, (1, 18, 3))
```

- better messages
- avoid issues with wrong index access if returned shape is 2D

Reply:
Ok, added new assertEqual